include/experimental/group/gemm/compute_policy.hpp

```diff
@@ -89,14 +88,68 @@ struct compute_policy_int4_dequantize<
 
   static constexpr uint32_t block_size_y_a = 16;
   using mma_attr = mma_attr_t<arch_tag_, block_size_y_a>;
-  static constexpr uint32_t block_bytes_x_a =
-      (mma_engine == mma_engine::xmx) ? mma_attr::mma_k_in_bytes : 32;
+  static constexpr uint32_t block_bytes_x_a = mma_attr::mma_k_in_bytes;
+  static constexpr uint32_t block_size_x_a =
+      block_bytes_x_a / sizeof(dtype_mma_a);
+  static constexpr uint32_t block_size_x_b = mma_attr::mma_n_in_elem;
+  static constexpr uint32_t block_bytes_y_b = mma_attr::mma_k_in_bytes;
+  static constexpr uint32_t block_size_y_b =
+      block_bytes_y_b / sizeof(dtype_mma_b);
+
+  static_assert(
+      block_bytes_x_a == block_bytes_y_b,
+      "mat_a x need to match with mat_b y");
+};
+
+/// @brief Specialized for fpu engine.
+template <
+    typename compute_attr_,
+    typename perf_tuning_knob_,
+    typename dtype_scale_,
+    typename dtype_zero_pt_,
+    quant_mode quant_type_,
+    int dequant_s_,
+    mma_engine mma_engine_,
+    gpu_arch arch_tag_>
+struct compute_policy_int4_dequantize<
+    compute_attr_,
+    perf_tuning_knob_,
+    dtype_scale_,
+    dtype_zero_pt_,
+    quant_type_,
+    dequant_s_,
+    mma_engine_,
+    arch_tag_,
+    std::enable_if_t<(
+        arch_tag_ <= gpu_arch::XeHpc && mma_engine_ == mma_engine::fpu)>> {
+  using compute_attr = compute_attr_;
+  using dtype_mma_acc = typename compute_attr::dtype_acc;
+  using dtype_mma_a = typename compute_attr::dtype_a;
+  using dtype_mma_b = typename compute_attr::dtype_b;
+
+  using perf_tuning_knob = perf_tuning_knob_;
+  static constexpr int stages = perf_tuning_knob::stages;
+  static constexpr int sync_freq = perf_tuning_knob::sync_freq;
+  static constexpr int k_stride = perf_tuning_knob::k_stride;
+  static constexpr mma_engine mma_engine = mma_engine_;
+  static constexpr gpu_arch arch_tag = arch_tag_;
+
+  static constexpr bool is_int4_matB_policy = true;
+
+  static constexpr uint32_t dequant_s = dequant_s_;
+  static_assert(
+      (dequant_s % (32 / sizeof(dtype_mma_b))) == 0,
+      "dequant_s should be a multiply of 32B");
+  using dtype_scale = dtype_scale_;
+  using dtype_zero_pt = dtype_zero_pt_;
+  static constexpr quant_mode quant_type = quant_type_;
+
+  static constexpr uint32_t block_size_y_a = 16;
+  static constexpr uint32_t block_bytes_x_a = 256;
   static constexpr uint32_t block_size_x_a =
       block_bytes_x_a / sizeof(dtype_mma_a);
-  static constexpr uint32_t block_size_x_b =
-      (mma_engine == mma_engine::xmx) ? mma_attr::mma_n_in_elem : 32;
-  static constexpr uint32_t block_bytes_y_b =
-      (mma_engine == mma_engine::xmx) ? mma_attr::mma_k_in_bytes : 32;
+  static constexpr uint32_t block_size_x_b = 32;
+  static constexpr uint32_t block_bytes_y_b = 256;
```

Comment:
why set block_bytes_y_b as so big ? and is that should be set for the different arch  ?
In pr255, I make this change as:
  using mma_attr = mma_attr_t<arch_tag_, mma_engine, block_size_y_a>;
  static constexpr uint32_t block_bytes_x_a = (mma_engine == mma_engine::xmx)
      ? mma_attr::mma_k_in_bytes
      : mma_attr::blk_k_in_bytes;
   static constexpr uint32_t block_size_x_a =
       block_bytes_x_a / sizeof(dtype_mma_a);
  static constexpr uint32_t block_size_x_b = (mma_engine == mma_engine::xmx)
      ? mma_attr::mma_n_in_elem
      : mma_attr::blk_n_in_bytes / sizeof(dtype_mma_b);
  static constexpr uint32_t block_bytes_y_b = block_bytes_x_a;
  static constexpr uint32_t block_size_y_b =
       block_bytes_y_b / sizeof(dtype_mma_b);

Reply:
Each block processes k-dimension 256/sizeof(fp16) 128 elements, which means it needs to load 128 int4, 64 bytes.

Reply:
In order to ensure the length of the load. In the end, the block size was very large

Reply:
https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_intel_esimd/sycl_ext_intel_esimd_functions.md#table1---valid-values-of-n-if-cache-hints-are-used-or-pred-parameter-is-passed   
for block1d max_vec_byte_len is 256 bytes, arch config max_vec_byte_len  is 32 may be wrong.
I will change it to 256byte and PVC 512